test/unit/pages/confirm/controller-test.js

```diff
@@ -1,31 +1,93 @@
+import InstallerDataService from 'browser/services/data';
+import InstallableItem from 'browser/model/installable-item';
+import ConfirmController from 'browser/pages/confirm/controller';
+
+require('../../../angular-test-helper');
+require('browser/main');
+
 chai.use(sinonChai);
 
 describe('ConfirmController', function() {
-  describe('initial state', function() {
-    beforeEach(ngModule('devPlatInstaller'));
 
-    var $controller;
+  let sandbox = sinon.sandbox.create();
+  let electron = new ElectronMock();
+  let $watch = sinon.stub();
+  let $scope = {$watch};
+  let $controller;
+  let $rootScope;
+  let confirmcontroller;
+  let installerDataSvc;
 
-    beforeEach(inject(function(_$controller_) {
+  describe('initial state', function() {
+    beforeEach(ngModule('devPlatInstaller'));
+    beforeEach(inject(function(_$controller_, _$rootScope_, _$state_) {
     // The injector unwraps the underscores (_) from around the parameter names when matching
       $controller = _$controller_;
+      $rootScope = _$rootScope_;
+      confirmcontroller = $controller('ConfirmController', {
+        $scope: $rootScope,
+        $state: _$state_,
+        $timeout: undefined,
+        installerDataSvc: new InstallerDataService(),
+        electron });
     }));
 
     it('installs watchers to track components selected for install', function() {
-      let $watch = sinon.stub();
-      let $scope = {$watch};
-      let ctrl = $controller('ConfirmController', { $scope });
+      expect($watch.callCount).to.be.equal(confirmcontroller.installerDataSvc.allInstallables().size+0);
+    });
+  });
+
+  beforeEach(function() {
+    sandbox.stub(Logger, 'getIpcRenderer').returns({send() {}});
+  });
 
-      expect($watch.callCount).to.be.equal(ctrl.installerDataSvc.allInstallables().size+3);
+  afterEach(function() {
+    sandbox.restore();
+  });
+
+  describe('back', function() {
+    beforeEach(function() {
+      sandbox.stub(confirmcontroller.router, 'go');
+      confirmcontroller.installerDataSvc.installRoot = 'folderName';
+      confirmcontroller.back();
+    });
+
+    it('sets selected folder as target folder in data service', function() {
+      expect(confirmcontroller.installerDataSvc.installRoot).to.be.equal('folderName');
+    });
+
+    it('navigates to location page', function() {
+      expect(confirmcontroller.router.go).calledWith('location');
+    });
+  });
+
+  describe('exit', function() {
+    it('exit closes active window', function() {
+      sandbox.stub(electron.remote.currentWindow);
+      confirmcontroller.exit();
+      expect(electron.remote.currentWindow.close).calledOnce;
     });
+  });
 
+  describe('download', function() {
+    it('should able to check download', function() {
+      expect(confirmcontroller.download('url')).to.be.equal(undefined);
+    });
+  });
+
+  describe('isConfigurationValid', function() {
+    it('should able to check Configuration Valid', function() {
```

Comment:
You mean `should BE able to check configuration IS valid` to be correct, but it still adds nothing to understanding what is checked. It should sound like specification:
* should return true if nothin is detected and all components have been selected for installation;
* should return true if all detected components have supported versions;
* should return false if one of detected components have unsupported version;
* you name it and get coverage as high as you can! :)